### PR TITLE
Add completion notification  to runner panel

### PR DESCRIPTION
**Pull request by LionelColaso**
Added Operation Completed Notification to runner_pannel to let user know its safe to close the window

Screenshot 2024-10-21 150928

image

### CODE REVIEW

**Comment by LionelColaso** [comment]
> Typo: "Operation" should not be capitalised.

its already capital ?

File: app/windows/runner_panel.py
Comment (hk21702) [inline app/windows/runner_panel.py:413]:
```suggestion
            text="Operation completed, you may close this window.",
```

**Review by hk21702 — CHANGES_REQUESTED** [review]

File: app/windows/runner_panel.py
Comment (LionelColaso) [reply app/windows/runner_panel.py:413]: i just used what i saw in other apps if u still prefer i will cghange "can" to "may"
image

File: app/windows/runner_panel.py
Comment (LionelColaso) [reply app/windows/runner_panel.py:411]: yeas sure any specifics you want to name the two buttons ?

Comment (hk21702) [reply app/windows/runner_panel.py:411]: See previous comment for example.

**Review by twstagg — APPROVED** [review]
Looks good to me. I approve pending @hk21702 approval.

It should be noted that this implementation only pops this message up when the runner panel has a `QProcess` associated, i.e. `steamcmd` or `todds`.

This does not address the original issue that was reported with DB Builder as it does not use the `QProcess`. We can address that later though. When DB Builder is finished, we will need to implement this sort of thing separately. I do need to finish making the process pool able to send messages back to the UI, so i can implement something accordingly when i complete that work.

File: app/windows/runner_panel.py
Comment (LionelColaso) [reply app/windows/runner_panel.py]: done

**Comment by LionelColaso** [comment]
> Looks good to me. I approve pending @hk21702 approval.
> 
> It should be noted that this implementation only pops this message up when the runner panel has a `QProcess` associated, i.e. `steamcmd` or `todds`.
> 
> This does not address the original issue that was reported with DB Builder as it does not use the `QProcess`. We can address that later though. When DB Builder is finished, we will need to implement this sort of thing separately. I do need to finish making the process pool able to send messages back to the UI, so i can implement something accordingly when i complete that work.

ohh i  thought  def finished(self)  was common for all

File: app/windows/runner_panel.py
Comment (hk21702) [inline app/windows/runner_panel.py:410]: Punctuation

File: app/windows/runner_panel.py
Comment (hk21702) [inline app/windows/runner_panel.py:407]: Just use a BinaryChoiceDialog or show_dialogue_conditional, omit the information part, and label the two buttons in a clear manner like the examples I gave before.

**Review by hk21702 — CHANGES_REQUESTED** [review]

File: app/windows/runner_panel.py
Comment (LionelColaso) [reply app/windows/runner_panel.py]: done

File: app/windows/runner_panel.py
Comment (LionelColaso) [reply app/windows/runner_panel.py:407]: done

File: app/windows/runner_panel.py
Comment (hk21702) [inline app/windows/runner_panel.py:412]: I think show_dialogue_conditional doesn't allow you to override the cancel button text. Try the following with BinaryChoiceDialog. Also, add a title and rename operation to process for consistency with the phrasing used in the runner panel logs.

```suggestion
        diag = BinaryChoiceDialog(
            title="Process Complete",
            text="Process complete, you can close the window.",
            positive_text="Close Window",
            negative_text="Ok",
        )
        if diag.exec_is_positive():
            self.close()
```

**Review by hk21702 — CHANGES_REQUESTED** [review]

File: app/windows/runner_panel.py
Comment (LionelColaso) [reply app/windows/runner_panel.py]: ok i just used your suggestion and done with it

also seems show_dialogue_input is depriciated ?
image

Comment (hk21702) [reply app/windows/runner_panel.py]: Yes. That is intended. It's just a wrapper around QInputDialog right now and doesn't add anything new.

**Review by hk21702 — APPROVED** [review]

**Pull request merged**
